### PR TITLE
Replace BuildPrediction code with the Microsoft.Build.Prediction package

### PR DESCRIPTION
**Pull request by dfederm**
The code now lives here: https://github.com/microsoft/MSBuildPrediction

The current version of the package is more or less just a port of the code from the repo, with very minor refactoring.

### CODE REVIEW

**Comment by dfederm** [comment]
Note that I'm waiting for the internal BuildXL.Selfhost.MyGet feed to upstream to the internal BuildXL.Selfhost feed, so the PR validation might fail until then...

File: config.dsc [hunk 1]
```diff
@@ -444,7 +443,9 @@ config({
                 // Extra dependencies to make MSBuild work
                 { id: "Microsoft.VisualStudio.Setup.Configuration.Interop", version: "1.16.30"},
                 { id: "System.CodeDom", version: "4.4.0"},
-                { id: "Microsoft.Build.Locator", version: "1.0.31"},
+
+                // Used for MSBuild input/output prediction
+                { id: "Microsoft.Build.Prediction", version: "0.1.0" },
```
Comment (dfederm) [inline config.dsc]: Removing at this package is unused. See `MsBuildAssemblyLoader.cs` for how we load MSBuild manually.

**Comment by dannyvv** [comment]
I sent github invite to join the BuildXL collaborators so the builds kick off automatically

---
In reply to: [498866672](https://github.com/microsoft/BuildXL/pull/411#issuecomment-498866672) [](ancestors = 498866672)

**Comment by dfederm** [comment]
For the failing status checks, all but 1 seem to only be running for me for some reason (not run for other PRs). The 1 is "[Optional]Primary BuildXL PR Validation(Schedule Builds Only)" which seems to be failing consistently for others and well, is marked as "optional". I'm going ahead and merging.

**Comment by smera** [comment]
> For the failing status checks, all but 1 seem to only be running for me for some reason (not run for other PRs). The 1 is "[Optional]Primary BuildXL PR Validation(Schedule Builds Only)" which seems to be failing consistently for others and well, is marked as "optional". I'm going ahead and merging.

Sounds good. Not sure why those are running only for you, to be honest.

**Pull request merged**